R/tags.R

```diff
@@ -928,6 +971,17 @@ as.tags.shiny.tag.list <- function(x, ...) {
   x
 }
 
+#' @export
+as.tags.shiny.tag.function <- function(x, ...) {
+  y <- x()
+  # as.tags() doesn't currently have a method for a list of dependencies
+  if (is.list(y) && all(vapply(y, inherits, logical(1), "html_dependency"))) {
```

Comment:
Will a case like the following be handled properly:

```R
div(list(
  div(),
  htmlDependency("abc", "1.0", "")
))
```

That is, where it's a plain list that has some, but not all, html dependencies.

Reply:
I'm not sure I understand the question. Are you concerned about this sort of example? This sort of thing should generally work because `tagify()` calls itself recursively on the return value of `as.tags()` (i.e., it's ok if `as.tags()` returns 'non-tag' objects because `as.tags()` will still get called on those contents)

```r
str(as.tags(tagFunction(function() {
  div(list(
    div(),
    htmlDependency("abc", "1.0", "")
  ))
})))
```

```r
List of 3
 $ name    : chr "div"
 $ attribs : Named list()
 $ children:List of 1
  ..$ :List of 2
  .. ..$ :List of 3
  .. .. ..$ name    : chr "div"
  .. .. ..$ attribs : Named list()
  .. .. ..$ children: list()
  .. .. ..- attr(*, "class")= chr "shiny.tag"
  .. ..$ :List of 10
  .. .. ..$ name      : chr "abc"
  .. .. ..$ version   : chr "1.0"
  .. .. ..$ src       :List of 1
  .. .. .. ..$ file: chr ""
  .. .. ..$ meta      : NULL
  .. .. ..$ script    : NULL
  .. .. ..$ stylesheet: NULL
  .. .. ..$ head      : NULL
  .. .. ..$ attachment: NULL
  .. .. ..$ package   : NULL
  .. .. ..$ all_files : logi TRUE
  .. .. ..- attr(*, "class")= chr "html_dependency"
 - attr(*, "class")= chr "shiny.tag"
```